### PR TITLE
Feature/add docker storage options

### PR DESCRIPTION
**Pull request by AblionGE**
### Description

Add the ability to specify docker storage options instead of having them hardcoded.

### CODE REVIEW

**Comment by AblionGE** [comment]
Please notice that this PR is started from https://github.com/puppetlabs/puppetlabs-kubernetes/pull/382

File: manifests/init.pp [hunk 1]
```diff
@@ -478,6 +486,11 @@
   Optional[String] $docker_yum_gpgkey                = undef,
   Optional[String] $docker_key_id                    = undef,
   Optional[String] $docker_key_source                = undef,
+  Optional[String] $docker_storage_driver            = 'overlay2',
```
Comment (cmccrisken-puppet) [inline manifests/init.pp]: Would there be any fall out from using `overlay2` if the OS does not support it? It appears as though you need to be at a specific version of the Linux kernel to be able to make use of it: https://docs.docker.com/storage/storagedriver/overlayfs-driver/

Comment (AblionGE) [reply manifests/init.pp]: Thanks for your comment @cmccrisken-puppet .
I rebased my branche to have only the commit related to the PR.

For the overlay2, I chose to be consistant with what is in master branch right now:
https://github.com/puppetlabs/puppetlabs-kubernetes/blob/master/templates/docker/daemon_redhat.json.erb#L11
https://github.com/puppetlabs/puppetlabs-kubernetes/blob/master/templates/docker/daemon_debian.json.erb#L11

But I can change it to force users to select their own storage driver. As you wish :)

Comment (cmccrisken-puppet) [reply manifests/init.pp]: Thanks for separating out the commit and pointing me to the template. Given we've defined it there, I'm happy enough to keep your change, in hindsight.

**Pull request merged**